src/mopsy/mops.py

```diff
@@ -77,3 +77,29 @@ def apply(
             raise Exception("ApplyFuncError")
 
         return result
+
+    def fapply(self, funcs: list, axis: int = 1) -> np.ndarray:
+        """ a reduction apply with multiple functions
+
+        Args:
+            funcs (list): functions to be called.
+            axis (int, optional): 0 for rows, 1 for columns. Defaults to 0.
+
+        Raises:
+            Exception: ApplyFuncError, when a function cannot be applied
+
+        Returns:
+            numpy.ndarray: a matrix
+        """
+        result = []
+        try:
+            for func in funcs:
+                for g, kmat in self.iter(group=None, axis=axis):
+                    tmat = kmat._apply(func, axis=axis)
+                result.append(tmat)
+            result = np.stack(result, axis=axis)
```

Comment:
its more efficient to iterate over groups and apply a set of functions on the slices

Reply:
change iteration to do over a dimension first